components/server/src/iam/iam-login-handler.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { injectable, inject } from "inversify";
+import * as express from "express";
+
+import { SessionHandlerProvider } from "../session-handler";
+import { UserService } from "../user/user-service";
+
+@injectable()
+export class IamLoginHandler {
+    @inject(SessionHandlerProvider) protected sessionHandlerProvider: SessionHandlerProvider;
+    @inject(UserService) protected readonly userService: UserService;
+
+    public async login(req: express.Request): Promise<SessionData> {
+        // TODO(at) create/update account based on id_token
+        // const idToken = req.body;
+
+        const user = await this.userService.createUser({
+            identity: {
+                authId: "fake-id-" + Date.now(),
+                authName: "FakeUser",
+                authProviderId: "oidc1",
+                primaryEmail: "fake@email.io",
+            },
+            userUpdate: (user) => {
+                user.name = "FakeUser";
+                user.fullName = "Fake User";
+                user.avatarUrl = "https://github.com/github.png";
+            },
+        });
+
+        await new Promise<void>((resolve, reject) => {
+            req.login(user, (err) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve();
+                }
+            });
```

Comment:
Do I read it correctly that with this change, we'd always get a rejection on the `req.login` because the user identity is a fake?

Reply:
The fake login is independent of the id_token at the moment. 

What it does, it creates a new user, and logs them in. It also returns the cookie. Unfortunately, I realized later that the session cookies name is missing 🤦🏻‍♂️ 

Why not guarding this? This cannot be called from outside, the single call site will be from OIDC flow, thus there is no uncontrolled call site.